homework/grayscale-image/grayscale.cpp

```diff
@@ -0,0 +1,56 @@
+#include "grayscale.hpp"
+
+CompressedBitmap compressGrayscale(const Bitmap& bitmap) {
+    CompressedBitmap compressed{};
+    compressed.reserve(width * height);
+
+    uint8_t currCount{1};
+    for (Bitmap::size_type i{0}; i < height; ++i) {
+        for (Bitmap::size_type j{1}; j < width + 1; ++j) {
+            while (j < width && bitmap[i][j] == bitmap[i][j - 1]) {
+                ++currCount;
+                ++j;
+            }
+            compressed.emplace_back(bitmap[i][j - 1], currCount);
+            currCount = 1;
+        }
+    }
+
+    compressed.shrink_to_fit();
+    return compressed;
+}
+
+Bitmap decompressGrayscale(const CompressedBitmap& compressed) {
+    Bitmap bitmap{};
+
+    auto it = bitmap.begin();
+    int count{0};
+    for (const auto& [Val, N] : compressed) {
+        std::fill_n(it->begin() + count, N, Val);
+        count += N;
+        if (count >= width) {
+            ++it;
+            count = 0;
+        }
```

Comment:
You can also read about array.fill(). It has special overload function for 2D array, which allow to fill more then one row :)